### ISSUE TITLE
Upload a binary file to Attachment field by using script?

### ISSUE THREAD

**Issue by lowcode101** [apitable/apitable#738]
Question:

I would like to upload a binary file, be it base64 string, or file some other website, how do I do that using script? 

I checked thru the APITable developer documents, it seems the property of Attachment, like URL is for those file already "uploaded"

**Comment by ziqiangai** [comment]
Hi @lowcode101 , Is the documentation [here](https://developers.apitable.com/api/upload-attachments) helpful?

Here is the api for uploading files.

**Comment by lowcode101** [comment]
Hi, thanks for the response, I think this is creating a widget? Can I make use of existing script widget to do that?

**Comment by lowcode101** [comment]
Hi, the example given also not showing the specific row to upload the attachment. I try to reference to this for updating datasheet: https://developers.apitable.com/script/examples/ip-location

**Comment by ziqiangai** [comment]
Yes, of course, you can call the API for uploading files in the script, but the problem is, how do you trigger the event of selecting files to upload?

**Comment by lowcode101** [comment]
Hi, I am trying to trigger within the widget. The widget will call a web service to generate a link to a PNG file, and I trying to store this PNG file on data sheet.

**Comment by mr-kelly** [comment]
As discussed in the thread, the Script Widget API currently focuses on data manipulation and does not natively expose a direct method for handling binary file uploads within its sandbox. To upload files, you need to use the standard REST API (`/api/v1/spaces/{spaceId}/nodes/{nodeId}/attachments`) via `fetch` within your script, and then write the returned attachment object back to the cell. We will close this issue for now, but your feedback on improving the Script Widget capabilities is noted!
